api/v1/kustomization_types.go

```diff
@@ -99,6 +99,20 @@ type KustomizationSpec struct {
 	// +optional
 	HealthChecks []meta.NamespacedObjectKindReference `json:"healthChecks,omitempty"`
 
+	// NamePrefix will prefix the names of all managed resources.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=200
+	// +kubebuilder:validation:Optional
```

Comment:
It seems that there isn't any difference between `// +kubebuilder:validation:Optional` and  `// +optional`

Reply:
This is how is set on others fields. If we decided to drop one, this is not the place to do it.